code/agents.py

```diff
@@ -64,13 +65,30 @@ def  __init__(self, position, velocity, fish_id, environment, ann, image = None,
     self._time_since_last_ANN_update
     self.fish_index
     """
-    
+
+    def get_fish_forces(self):
+        f = np.array([0.0, 0.0])
+        neighbours = self.get_neighbours()
```

Comment:
Assume neigbours are already found by think(), just get them directly from the attribute self.neighbouring_fish. That way we don't have to do the search twice.